main.c

```diff
@@ -32,6 +32,14 @@ int main(int argc, char** argv) {
             algorithm_runner.start = search_best_fit_runner.start;
         }
 
+        if(strcmp(algorithm, "first_fit") == 0) {
+            algorithm_runner.start = search_first_fit_runner.start;
+        }
+
+        if(strcmp(algorithm, "next_fit") == 0) {
+            algorithm_runner.start = search_next_fit_runner.start;
+        }
+
         if(algorithm_runner.start != NULL) {
             algorithm_runner.start(chunks_fs);
```

Comment:
here this seems not to accept `sizes_fs` which is also crucial part of the inout (my bad tho)